src/IO/ReadHelpers.cpp

```diff
@@ -336,7 +336,23 @@ static ReturnType parseComplexEscapeSequence(Vector & s, ReadBuffer & buf)
         ++buf.position();
         /// escape sequence of the form \xAA
         char hex_code[2];
-        readPODBinary(hex_code, buf);
+
+        if constexpr (std::is_same_v<ReturnType, void>)
+        {
+            readPODBinary(hex_code, buf);
+        }
+        else
+        {
+            try
+            {
+                readPODBinary(hex_code, buf);
+            }
+            catch (const Exception &)
```

Comment:
Hm, that is an interesting question. I initially thought it was an error when I saw that `parseComplexEscapeSequence<String, bool>` could throw even though `ReturnType` is set not to throw. I don't quite remember if there were side effects of not catching all exceptions, tho.

Basically, it's a matter of design / expectations about the function. Do you have a strong opinion on this?

Reply:
I believe there're exceptions that we don't want to catch. ReadBuffer can be reader from remote source and it can be network error. Or error on memory allocation.